File: DotNetGotchas/CSharp/NotToFinalize/Modified1/FinalizeAndManagedResources/SomeClass.cs
//SomeClass.cs
using System;

namespace FinalizeAndManagedResources
{
	public class SomeClass
	{
		private SomeOtherClass other;

		public void Setother(SomeOtherClass otherObject)
		{
			other = otherObject;
		}

		~SomeClass()
		{
			Console.WriteLine("Finalize called on SomeClass");
			if(other != null)
			{
				other.Notify();
			}
		}
	}
}

File: DotNetGotchas/VBDN/NotToFinalize/Modified1/FinalizeAndManagedResources/SomeClass.vb
'SomeClass.vb
Public Class SomeClass
    Private other As SomeOtherClass

    Public Sub Setother(ByVal otherObject As SomeOtherClass)
        other = otherObject
    End Sub
    Protected Overrides Sub Finalize()
        MyBase.Finalize()
        Console.WriteLine("Finalize called on SomeClass")
        If Not other Is Nothing Then
            other.Notify()
        End If
    End Sub
End Class